cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ReplicaDistributionGoal.java

```diff
@@ -204,8 +204,9 @@ private boolean rebalanceByMovingReplicasOut(Broker broker,
 
     // Get the replicas to rebalance. Replicas are sorted from smallest to largest disk usage.
     List<Replica> replicasToMove = broker.trackedSortedReplicas(name()).sortedReplicas();
-    // If cluster has dead brokers, but this broker is alive, then limit moving replica to immigrant replica.
-    if (!clusterModel.deadBrokers().isEmpty() && broker.isAlive()) {
+    // If (1) cluster has dead brokers, but this broker is alive, or (2) optimization option requests so,
+    // then limit moving replica to immigrant replica.
+    if ((!clusterModel.deadBrokers().isEmpty() && broker.isAlive()) || optimizationOptions.onlyMoveImmigrantReplicas()) {
```

Comment:
Should `initGoalState` add the following `selectionFunc` as parameter as in capacity and resource distribution goals?

```
optimizationOptions.onlyMoveImmigrantReplicas() ? ReplicaSortFunctionFactory.selectImmigrants() : null,
```